api/http.go

```diff
@@ -59,34 +62,35 @@ type upgradeResponse struct {
 	Data   string `json:"data,omitempty"`
 }
 
-// listRequest is the json schema for the list api
-type listRequest struct {
-	flags.ListFlags
-}
-
 // listResponse is the json schema to parse the list api response
 type listResponse struct {
 	Error    string            `json:"error,omitempty"`
 	Releases []release.Release `json:"releases,omitempty"`
 }
 
 // request is a helper function to append the path to baseUrl and send the request to the APIClient
-func (c *HttpClient) request(ctx context.Context, reqPath string, method string, body io.Reader) (*http.Response, []byte, error) {
+func (c *HttpClient) request(ctx context.Context, reqPath string, method string, body io.Reader, queryString string) (*http.Response, []byte, error) {
 	u := *c.baseUrl
 	u.Path = path.Join(strings.TrimRight(u.Path, "/"), reqPath)
+	u.RawQuery = queryString
 	return c.client.Send(u.String(), method, body)
 }
 
 // List sends the list api request to the APIClient and returns a list of releases if successfull.
 func (c *HttpClient) List(ctx context.Context, fl flags.ListFlags) ([]release.Release, error) {
-	reqBody, err := json.Marshal(&listRequest{
-		ListFlags: fl,
-	})
+	reqPath, err := getListPath(fl.KubeContext, fl.Namespace, fl.AllNamespaces)
 	if err != nil {
 		return nil, err
 	}
-
-	_, data, err := c.request(ctx, "list", http.MethodGet, bytes.NewBuffer(reqBody))
+	queryParams := url.Values{}
+	err = encoder.Encode(fl, queryParams)
+	if err != nil {
+		return nil, err
+	}
+	httpResponse, data, err := c.request(ctx, reqPath, http.MethodGet, nil, queryParams.Encode())
+	if httpResponse.StatusCode == 204 {
+		return []release.Release{}, nil
+	}
 	if err != nil {
```

Comment:
err !=nil should be the first thing post calling func, as at times response would be nil